crates/biome_js_analyze/src/analyzers/nursery/use_shorthand_function_type.rs

```diff
@@ -0,0 +1,276 @@
+}
+
+impl Rule for UseShorthandFunctionType {
+    type Query = Semantic<Query>;
+    type State = RuleState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let query = ctx.query();
+
+        match query {
+            Query::TsInterfaceDeclaration(interface_declaration) => {
+                if interface_declaration.members().len() == 1
+                    && interface_declaration.extends_clause().is_none()
+                {
+                    if let Some(TsCallSignatureTypeMember(call_signature)) =
+                        interface_declaration.members().first()
+                    {
+                        return Some(RuleState::TsInterfaceDeclaration(
+                            query.range(),
+                            call_signature.parameters().ok()?.items(),
+                            call_signature.return_type_annotation()?.ty(),
+                        ));
+                    }
+                }
+                None
+            }
+            Query::JsFormalParameter(parameter) => {
+                if let Some(TsObjectType(ts_object)) = parameter.type_annotation()?.ty().ok() {
+                    if ts_object.members().len() == 1 {
+                        if let Some(ts_call_signature_type_member) = ts_object
+                            .members()
+                            .first()?
+                            .as_ts_call_signature_type_member()
+                        {
+                            return Some(RuleState::TsObjectType(
+                                ts_object.range(),
+                                ts_call_signature_type_member.parameters().ok()?.items(),
+                                ts_call_signature_type_member
+                                    .return_type_annotation()?.ty(),
+                            ));
+                        }
+                    }
+                }
+                None
+            }
+        }
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let (range, message, note) = match state {
+            RuleState::TsInterfaceDeclaration(range, parameters, return_type_syntax) => (
+                range,
+                markup! {
+                    "Prefer function type over interface."
+                },
+                markup! {"Interface only has a call signature, you should use a function type instead."},
+            ),
+            RuleState::TsObjectType(range, parameters, return_type_syntax) => (
+                range,
+                markup! {
+                    "Prefer function type over object type."
+                },
+                markup! {"Object only has a call signature, you should use a function type instead."},
+            ),
+        };
+        Some(RuleDiagnostic::new(rule_category!(), range, message).note(note))
```

Comment:
i think we can use the same diagnostic for interface and object type. What do you think?

```suggestion
        Some(RuleDiagnostic::new(rule_category!(), ctx.query().range(), markup! {
            "Use a function type instead of a call signature."
        }).note(markup { "Types containing only a call signature can be shortened to a function type." }))
```